### PR TITLE
feat: add advisors for self hosted

### PR DESCRIPTION
**Pull request by tranhoangvuit**
## What kind of change does this PR introduce?

feature: We enabled the advisor for self-hosted environments, which will provide security and performance advice.

## What is the current behavior?

Now it only available on cloud service.

## What is the new behavior?

We enabled advisors function for self hosted.

## Additional context

This is some screenshot about this feature:

image

image

### CODE REVIEW

File: apps/studio/pages/api/projects/[ref]/run-lints.ts [hunk 1]
```diff
@@ -2,6 +2,9 @@ import { NextApiRequest, NextApiResponse } from 'next'
 
 import apiWrapper from 'lib/api/apiWrapper'
 import { extractResponse } from 'pages/api/constants'
+import { constructHeaders } from 'lib/api/apiHelpers'
+import { post } from 'lib/common/fetch'
```
Comment (saltcod) [inline apps/studio/pages/api/projects/[ref]/run-lints.ts:6]: can we swap these from the updated fetchers?

https://share.cleanshot.com/lZF17fL4

**Comment by saltcod** [comment]
Does the lint-query 
https://github.com/supabase/supabase/blob/master/apps/studio/data/lint/lint-query.ts#L18

need the API_URL constant so it knows to swap the api url? 
https://github.com/supabase/supabase/blob/master/apps/studio/lib/constants/index.ts#L5

**Comment by saltcod** [comment]
Our platform api also runs these checks: 

const authConfigLintResults = []
if (authConfigs.EXTERNAL_EMAIL_ENABLED) {
      if (authConfigs.MAILER_OTP_EXP > 3600) {
        authConfigLintResults.push({
          name: 'auth_otp_long_expiry' as const,
          title: 'Auth OTP long expiry',
          level: 'WARN' as const,
          facing: 'EXTERNAL',
          categories: ['SECURITY' as const],
          description: 'OTP expiry exceeds recommended threshold',
          detail:
            'We have detected that you have enabled the email provider with the OTP expiry set to more than an hour. It is recommended to set this value to less than an hour.',
          cache_key: 'auth_otp_long_expiry',
          remediation: 'https://supabase.com/docs/guides/platform/going-into-prod#security',
          metadata: {
            type: 'auth' as const,
            entity: 'Auth',
          },
        })
      }

      if (authConfigs.EXTERNAL_PHONE_ENABLED && authConfigs.SMS_OTP_LENGTH < 6) {
        authConfigLintResults.push({
          name: 'auth_otp_short_length' as const,
          title: 'Auth OTP short length',
          level: 'WARN' as const,
          facing: 'EXTERNAL',
          categories: ['SECURITY' as const],
          description: 'OTP length is less than recommended threshold',
          detail: 'We have detected that you have set the OTP length to less than 6 characters',
          cache_key: 'auth_otp_short_length',
          remediation: 'https://supabase.com/docs/guides/platform/going-into-prod#security',
          metadata: {
            type: 'auth' as const,
            entity: 'Auth',
          },
        })
      }
    }
    
    returning a single array with these ^^ plus any results from the big lint query 

like this: 
`return [...splinterLintResults, ...authConfigLintResults]`

File: apps/studio/pages/api/projects/[ref]/run-lints.ts (hunk 1 above)
Comment (tranhoangvuit) [reply apps/studio/pages/api/projects/[ref]/run-lints.ts:6]: Hi @saltcod , based on [this discussion](https://supabase.slack.com/archives/C03S3SW9485/p1718894630949059?thread_ts=1718893216.484629&cid=C03S3SW9485), we can keep using `post` from `lib/common/fetch`.

**Comment by tranhoangvuit** [comment]
>Does the lint-query
>https://github.com/supabase/supabase/blob/master/apps/studio/data/lint/lint-query.ts#L18

>need the API_URL constant so it knows to swap the api url?
>https://github.com/supabase/supabase/blob/master/apps/studio/lib/constants/index.ts#L5
we don't need `API_URL`, because default we checked with `IS_PATHFORM` for ``PG_META_URL` already.


> Our platform api also runs these checks:
> 
> ```
> if (authConfigs.EXTERNAL_EMAIL_ENABLED) {
>       if (authConfigs.MAILER_OTP_EXP > 3600) {
>         authConfigLintResults.push({
>           name: 'auth_otp_long_expiry' as const,
>           title: 'Auth OTP long expiry',
>           level: 'WARN' as const,
>           facing: 'EXTERNAL',
>           categories: ['SECURITY' as const],
>           description: 'OTP expiry exceeds recommended threshold',
>           detail:
>             'We have detected that you have enabled the email provider with the OTP expiry set to more than an hour. It is recommended to set this value to less than an hour.',
>           cache_key: 'auth_otp_long_expiry',
>           remediation: 'https://supabase.com/docs/guides/platform/going-into-prod#security',
>           metadata: {
>             type: 'auth' as const,
>             entity: 'Auth',
>           },
>         })
>       }
> 
>       if (authConfigs.EXTERNAL_PHONE_ENABLED && authConfigs.SMS_OTP_LENGTH < 6) {
>         authConfigLintResults.push({
>           name: 'auth_otp_short_length' as const,
>           title: 'Auth OTP short length',
>           level: 'WARN' as const,
>           facing: 'EXTERNAL',
>           categories: ['SECURITY' as const],
>           description: 'OTP length is less than recommended threshold',
>           detail: 'We have detected that you have set the OTP length to less than 6 characters',
>           cache_key: 'auth_otp_short_length',
>           remediation: 'https://supabase.com/docs/guides/platform/going-into-prod#security',
>           metadata: {
>             type: 'auth' as const,
>             entity: 'Auth',
>           },
>         })
>       }
>     }
>     ```
>     returning a single array with these ^^ plus any results from the big lint query 
> 
> like this: 
> `return [...splinterLintResults, ...authConfigLintResults]`
> 
> let me know if this part doesn't make sense!
> ```
Thank you so much for your suggestion, but we have some issues here. We call the [auth AP](https://github.com/supabase/supabase/blob/master/apps/studio/pages/api/auth/%5Bref%5D/config.ts#L24)I to get the settings from there, but actually, in the settings of [GoTrue](https://github.com/supabase/auth/blob/master/internal/api/settings.go#L43), we don't return `MAILER_OTP_EXP` or `SMS_OTP_LENGTH`, so it's hard to check here. Should we skip it for now, or can we ask the auth team to return more values in the settings?

**Review by saltcod — APPROVED** [review]

**Comment by saltcod** [comment]
Looks good, will merge this tomorrow morning. Thanks @tranhoangvuit !

**Comment by saltcod** [comment]
checked preview

**Comment by tranhoangvuit** [comment]
Thank you @saltcod for helping me update it.

**Pull request merged**